pkg/buildah/login.go

```diff
@@ -59,14 +67,51 @@ func newLoginCommand() *cobra.Command {
 		Use:   "login",
 		Short: "Login to a container registry",
 		Long:  loginDescription,
+
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			if opts.kubeconfig == "" {
+				return
+			} else {
+				// set user/password
+				var passwordb []byte
+				if passwordb, err = fileutil.ReadAll(opts.kubeconfig); err != nil {
+					return err
+				}
+				// username is current context kubeconfig user id
+				username, err := GetCurrentUserFromKubeConfig(opts.kubeconfig)
+				if err != nil {
+					return err
+				}
+				opts.loginOpts.Username = username
+				opts.loginOpts.Password = string(passwordb)
+
+				// config will be copyed to $(HOME)/.sealos/$(args[0]).HOST/$(user)/config
+				registryHost, err := replaceURLByHostPort(args[0])
+				if err != nil {
+					return err
+				}
+				sealosKubeConfdir := fmt.Sprintf("%s/%s/%s/%s", os.Getenv("HOME"), ".sealos", registryHost, username)
+				err = fileutil.MkDirs(sealosKubeConfdir)
+				if err != nil {
+					return err
+				}
+				sealosKubeconfPath := fmt.Sprintf("%s/%s", sealosKubeConfdir, "config")
+				err = fileutil.Copy(opts.kubeconfig, sealosKubeconfPath)
```

Comment:
Seems we must wait login command succeeded, and then do the real copy action?

Reply:
I will move it to PostRunE.